tests/helpers/test_cherrypicker_data_helper.py

```diff
@@ -0,0 +1,320 @@
+    barcode = "TEST-123456"
+
+    def create_row_side_effect(dt_arg, _, result_arg, barcode_arg):
+        nonlocal positives, negatives, dt, barcode
+
+        assert dt_arg == dt
+        assert barcode_arg == barcode
+
+        if result_arg == "Positive":
+            positives += 1
+        elif result_arg == "Negative":
+            negatives += 1
+
+        return ["A", "row"]
+
+    create_row.side_effect = create_row_side_effect
+
+    actual = create_plate_rows(dt, 40, barcode)
+    expected = [["A", "row"]] * 96
+
+    assert actual == expected
+    assert shuffle.called_with(["Positive"] * 40 + ["Negative"] * 56)
+    assert create_row.call_count == 96
+    assert positives == 40
+    assert negatives == 56
+
+
+def test_flat_list_of_positives_per_plate():
+    actual = flat_list_of_positives_per_plate([[2, 5], [3, 10]])
+    expected = [5, 5, 10, 10, 10]
+
+    assert actual == expected
+
+
+def test_create_csv_rows():
+    # Note this is an integration of all the methods tested above, so not strictly a unit test!
+    plate_specs = [[1, 0], [2, 40], [1, 96]]
+    dt = datetime(2012, 3, 4, 5, 6, 7)
+    barcodes = {"TEST-00POS01": 0, "TEST-40POS01": 40, "TEST-40POS02": 40, "TEST-96POS01": 96}
+
+    actual = create_csv_rows(plate_specs, dt, list(barcodes.keys()))
+
+    wells_per_plate = 96
+    expected_count = wells_per_plate * len(barcodes)
+    assert len(actual) == expected_count
+
+    # Check that identifier fields are unique across the rows
+    assert len(set([row[0] for row in actual])) == expected_count
+    assert len(set([row[1] for row in actual])) == expected_count
+    assert len(set([row[2] for row in actual])) == expected_count
+    assert len(set([row[3] for row in actual])) == expected_count
+
+    # Check that the timestamp was added to all rows identically
+    assert len(set([row[5] for row in actual])) == 1
+
+    # Per plate checks
+    for barcode, positives in barcodes.items():
+        barcode_rows = [row for row in actual if barcode in row[0]]
+        assert len(barcode_rows) == wells_per_plate
+
+        # Assert that expected fields contain the correct prefix and barcode
+        assert all([row[0].startswith(f"RSID-{barcode}") for row in barcode_rows])
+        assert all([row[1].startswith(f"VPID-{barcode}") for row in barcode_rows])
+        assert all([row[2].startswith(f"{barcode}_") for row in barcode_rows])
+        assert all([row[3].startswith(f"RNA_PCR-{barcode}") for row in barcode_rows])
+
+        # Check the correct number of positives and negatives were generated
+        positive_rows = [row for row in barcode_rows if "Positive" == row[4]]
+        negative_rows = [row for row in barcode_rows if "Negative" == row[4]]
+        assert len(positive_rows) == positives
+        assert len(negative_rows) == 96 - positives
+
+
+@pytest.fixture
+def test_rows_data():
+    return [
+        ["RSID-01", "VPID-01", "RNAID-01", "RNAPCRID-01", "Positive", "Timestamp", "AP"] + [""] * 12,
```

Comment:
This doesn't look realistic, the ids should contain plate barcodes and the well numbers are padded to 4 digits, and RNA_ID is <plate_barcode>_<well_coordinate> e.g AP-chp-12345678_A01
But maybe ok if you are just using strings for the tests?

Reply:
It shouldn't make any difference.  It's only here as something for the mock to return and to check that it got added verbatim the CSV rows later.